### DISCUSSION TITLE
Updating an existing statamic entry with new data

### DISCUSSION THREAD

**Discussion by RohanKambleDev** [statamic/cms#4910] [Q&A] [answered]
I have an existing entry in my Pages collection in statamic.

<img width="771" alt="Screenshot 2021-12-15 at 7 44 58 AM" src="https://user-images.githubusercontent.com/968285/146111426-e6b2fb7b-ec3a-4636-8293-662516f8d0b6.png">

I want to add / append a new item to that exisiting entry in "add_videos" array under block of type "video_block" as highlighted in the screenshot from my core Laravel 

So far my code looks like this

I fetched the exiting entry

`$exquisitelyHuman = Entry::query()
                        ->where('collection', 'pages')
                        ->where('slug', 'exquisitely-human')
                        ->first();`

this is the new video set I would like to add / append to the existing one

`$videoSet = [
                "title" => "Entry from Laravel Test Rohan",
                "description" => "Entry from Laravel description Test Rohan",
                "youtube_url" => "https://www.youtube.com/watch?v=F_skvaSPhI4",
                "type" => "new_set",
                "enabled" => true
            ];`

Now the only problem is I am not able to add the new data to that 'video_block' instead it gets added outside at the root level using this 

`$exquisitelyHuman->fromWorkingCopy()
                ->set('add_videos', $videoSet)
                ->save();`
(I referred this from Duncan's article - https://duncanmcclean.com/building-a-likes-addon-in-statamic-3)

is there a way where I can get hold of that data item and append a new item to it?

**Comment by jasonvarga** [comment, marked as answer]
The field you need to modify is the top level `blocks`.

You'll need to get the value of that, modify it, and then re-set it.

```php
$blocks = $entry->get('blocks');

// I don't know how you're picking which block to update.
// If it's just the first "video_block" block, then this will do it.
$index = collect($blocks)->search(function ($block) {
    return $block['type'] === 'video_block';
});

$blocks[$index]['add_videos'][] = ['title' => 'Entry from Laravel Test Rohan'];

$entry->set('blocks', $blocks);
$entry->save();
```

**Reply by RohanKambleDev** [reply]
Hey @jasonvarga Thanks a lot, it worked.
I just love Statamic Support be it on discord or github, You guys rock.
